pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -396,10 +386,12 @@ func (exp *flowExporter) addRecordToSet(record flowexporter.FlowRecord) error {
 		case "flowEndSeconds":
 			ie.Value = uint32(record.Conn.StopTime.Unix())
 		case "flowEndReason":
-			if record.IsActive {
-				ie.Value = activeTimeoutReason
+			if record.Conn.TCPState == "TIME_WAIT" || record.Conn.TCPState == "CLOSE" {
+				ie.Value = ipfixregistry.EndOfFlowReason
```

Comment:
Even though RFC states "EndOfFlowReason" is for TCP FIN, I feel we can use it for non- TCP protocol flows with the dying status flag indicating termination. Below is the statement from ipfix registry doc.

>The Flow was terminated because the Metering Process detected signals indicating the end of the Flow, for example, the TCP FIN flag.

@antoninbas Do you think we should stick "IdleFlowReason" for non-TCP protocol flows that end with the dying status flag or moving to "EndOfFlowReason" is ok?

Reply:
Moving to `EndOfFlowReason` sounds fine to me. I am not sure what triggers a connection to go into dying state. If it's based on a timeout, our idle timeout is likely to be much smaller than the conntrack timeout, so there we will send an `IdleFlowReason` first anyway.  If it's because the flow is explicitly removed from conntrack (I think it's common to do that for a Pod's UDP connections when the Pod is deleted, although I don't think we do it in Antrea), then it makes sense to send an `EndOfFlowReason`.

BTW, what happens in the following scenario: TCP connection goes idle for a minute, then goes into the `TIME_WAIT` state. Will we send an `IdleFlowReason` record first, followed by an `EndOfFlowReason` record after a minute?

Reply:
Yes, I considered timeout-based removal. We will send an `IdleFlowReason` record first and then `EndOfFlowReason` later. `EndOfFlowReason` could help processing at IPFIX collector as it indicates termination of a flow.
For explicit removal of the conntrack connection, `EndOfFlowReason` definitely makes sense. Thanks for the insight.

Yes, my expectation is the same. We should send `IdleFlowReason` record first and then `EndOfFlowReason` record. However, we should make sure that `record.IsActive` flag tagged as active. The current logic is that if there is any change in stats, we tag the flag back as active. We should extend the logic to state change as well. @zyiou could you make sure of that?

Reply:
Sure. you mean that when the state changes, we tag the IsActive flag back to active, right?

Reply:
Yes, that is correct. That will let us send the final record and delete the record in record map.